### PR TITLE
hash long cache keys

### PR DESCRIPTION
**Pull request by Adrian2112**
Some cache stores have a size limit for the keys. E.g ActiveSupport::Cache::FileStore
So in case there are a lot of parameters and the cache key is too long we hash it

### CODE REVIEW

File: lib/cached_resource/caching.rb [hunk 1]
```diff
@@ -112,7 +112,10 @@ def cache_clear
 
       # Generate the request cache key.
       def cache_key(*arguments)
-        "#{name.parameterize.gsub("-", "/")}/#{arguments.join('/')}".downcase.delete(' ')
+        arguments_string = arguments.join('/')
+        arguments_key = arguments_string.length > 150 ? Digest::SHA2.hexdigest(arguments_string) : arguments_string
```
Comment (mhgbrown) [inline lib/cached_resource/caching.rb]: Could you constantize this "150" somewhere?

**Comment by mhgbrown** [comment]
Also would appreciate if you could create a test for this

**Comment by Daniel-ltw** [comment]
@Adrian2112 could you bump the version for this change?

**Comment by Adrian2112** [comment]
I just rebased agains master. Does that work?

**Comment by Daniel-ltw** [comment]
Could this be optional? 

Setup a configuration variable for it. 
Only hash if the configuration is set to true and could probably default to true on ActiveSupport::Cache::FileStore?

**Comment by jlurena** [comment]
Closing for https://github.com/mhgbrown/cached_resource/pull/72

**Pull request closed without merging**